cookbook/case_studies/ml_training/nlp_processing/word2vec_and_lda.py

```diff
@@ -0,0 +1,331 @@
+# from word IDs to words as input to the LDA model for training. The words in each topic are printed
+# out once the model has finished training. Finally, the model is serialised to disk.
+@task
+def train_lda_model(
+    corpus: List[List[str]], hyperparams: LDAModelHyperparams
+) -> model_file:
+    id2word = Dictionary(corpus)
+    bow_corpus = [id2word.doc2bow(doc) for doc in corpus]
+    id_words = [[(id2word[id], count) for id, count in line] for line in bow_corpus]
+    print(f"sample of bag of words generated: {id_words[:2]}")
+    lda = LdaModel(
+        corpus=bow_corpus,
+        id2word=id2word,
+        num_topics=hyperparams.num_topics,
+        alpha=hyperparams.alpha,
+        passes=hyperparams.passes,
+        chunksize=hyperparams.chunksize,
+        update_every=hyperparams.update_every,
+        random_state=hyperparams.random_state,
+    )
+    pprint(f"Shows words in each topic: {lda.print_topics(num_words=5)}")
+    file = "lda.model"
+    lda.save(file)
+    return (file,)
+
+
+# %%
+# Word Similarities
+# ========
+#
+# The model is deserialised from disk and used to compute the top 10 most similar
+# words in a corpus to a given word (we will use the word `computer` when running
+# the workflow to output the similar words). Note that since the model is trained
+# on a small corpus, some of the relations might not be so clear.
+@task(cache_version="1.0", cache=True, limits=Resources(mem="200Mi"))
+def word_similarities(model_ser: FlyteFile[MODELSER_NLP], word: str):
+    model = Word2Vec.load(model_ser.path)
+    wv = model.wv
+    print(f"Word vector for {word}:{wv[word]}")
+    print(f"Most similar words in corpus to {word}: {wv.most_similar(word, topn=10)}")
+
+
+# %%
+# Sentence Similarity
+# ========
+#
+# This task computes the Word Mover’s Distance (WMD) metric using the trained embeddings of words. This
+# enables us to assess the distance between two documents in a meaningful way even when they have
+# no words in common. WMD is larger for two completely unrelated sentences and smaller for two closely related
+# sentences. We have already chosen two similar sentences for comparison so the word movers distance
+# should be a low value. You can try altering either ``SENTENCE_A`` or ``SENTENCE_B`` variables to be dissimilar
+# to the other sentence, and check that the value computed is larger.
+@task(cache_version="1.0", cache=True, limits=Resources(mem="200Mi"))
+def word_movers_distance(model_ser: FlyteFile[MODELSER_NLP]) -> float:
+    sentences = [SENTENCE_A, SENTENCE_B]
+    results = []
+    for i in sentences:
+        result = [w for w in utils.tokenize(i) if w not in STOPWORDS]
+        results.append(result)
+    model = Word2Vec.load(model_ser.path)
+    distance = model.wv.wmdistance(*results)
+    print(f"Computing word movers distance for: {SENTENCE_A} and {SENTENCE_B} ")
+    print(f"Word Movers Distance is {distance}")
```

Comment:
Can you use logging instead of print? https://docs.flyte.org/projects/cookbook/en/latest/auto/core/containerization/raw_container.html